### PR TITLE
Support path computation strategy param in partial flow update

### PR DESCRIPTION
**Pull request by rozdy**
Related to #2894

### CODE REVIEW

**Comment by rozdy** [comment]
> 
> > ```java
> >     /**
> >      * Updates max latency, priority or path computation strategy of existing flow.
> > ```
> 
> AS for me - list all modifiable fields here is bad idea.

Fixed

> 
> > ```java
> >      * Convert {@link String} to {@link PathComputationStrategy}.
> >      */
> >     default PathComputationStrategy mapPathComputationStrategy(String pathComputationStrategy) {
> > ```
> 
> I believe mapstruct is able to produce this mapper by it's own.

Yep but auto generated one is case-sensitive.

**Comment by sergii-iakovenko** [comment]
Please, address the review comments.

**Comment by rozdy** [comment]
> What's the point in this pr? If operator changes pce strategy for the flow, it's expected to be applied right now, which means -> new path computation, and the heavy update. Does it make sense to overwrite actual state this way?

Reworked patch. Introduced target path computation param for partial update.

**Comment by rozdy** [comment]
@sergii-iakovenko 
> This PR contains multiple changes of just renaming pathComputationStrategy -> targetPathComputationStrategy. What is the purpose of this? Why do we need to have targetPathComputationStrategy in the model's Flow?

The idea behind this PR is to provide ability to change flow path computation strategy without immediate reroute. New strategy (called "target") should be used with next reroute or full flow update. Old strategy is saved to provide consistency between `pathComputationStrategy` flow parameter and strategy used for current path computation. So a new field `targetPathComputationStrategy` is introduced. From API perspective old one should be considered as read-only flow param and new one should be used in payloads for flow update/patch.

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/update/actions/UpdateFlowAction.java [hunk 1]
```diff
@@ -82,8 +82,7 @@ public UpdateFlowAction(PersistenceManager persistenceManager, int transactionRe
     }
 
     private RequestedFlow updateFlow(Flow flow, RequestedFlow targetFlow, FlowUpdateFsm stateMachine) {
-        RequestedFlow originalFlow = RequestedFlowMapper.INSTANCE.toRequestedFlow(flow);
-        stateMachine.setOriginalFlow(originalFlow);
+        stateMachine.setOriginalFlow(RequestedFlowMapper.INSTANCE.copy(flow));
```
Comment (dpoltavets) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/update/actions/UpdateFlowAction.java]: Need to check that this change will not return the already resolved bugs associated with multi table.

Comment (rozdy) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/update/actions/UpdateFlowAction.java]: Fixed

**Pull request merged**